docs/developer/validation-rules/resource-model-validation.md

```diff
@@ -0,0 +1,38 @@
+# Identifying resource model definitions in a specification
+## Description
+A model definition is considered as a resource model definition if:
+
+- For each model, check if that model is returned by a PUT operation as a part of 200 or 201 response codes.
+- If the model has an 'allOf' Resource at any level [name of the root model is 'Resource'] AND the model is returned by a GET operation in 200 response.
+- If the model has 'x-ms-azure-resource': true or if any model has an 'x-ms-azure-resource' set to true in its model hierarchy and is not named either of 'Resource', 'TrackedResource' or 'ProxyResource', consider this as a resource
+
+## Types of resource models
+1. TrackedResource: A resource is considered a tracked resource if a model hierarchy has a property named 'location' which is marked as required.
+2. ProxyResource: Proxy resources are all resources without a required 'location' property in their model hierarchy
+
+## Validation rules for resource models
+
+### PutGetPatchResponseInvalid 
```

Comment:
should we split each of this in separate files? and add links to those from here? This way we'd be more consistent with https://github.com/Azure/autorest/commit/e225e61e36644ba9119aa4996e2d3554cbe2c78c and we can add more details to each page as needed.